### ISSUE TITLE
Role of corOutCoverage in canu1.7

### ISSUE THREAD

**Issue by johnomics** [marbl/canu#923]
canu 1.7 "Ensure[s] that every raw read is either corrected or used as evidence for correcting some other raw read. This serves to rescue short plasmids in high coverage datasets, and it is no longer necessary to set corOutCoverage to achieve the same result." (https://github.com/marbl/canu/releases/tag/v1.7); it also "dynamically select poorly represented sequences to avoid missing short plasmids" (http://canu.readthedocs.io/en/latest/faq.html).

Could you expand on this a little please? corOutCoverage still exists, so what is its use now? When might it be increased? If it no longer influences which reads are retained for correction, is it still used to assess reads for trimming, unitigging, consensus etc?

Also, is there some way to tweak how the dynamic selection of poorly represented sequences works? We are seeing quite a few small contigs being retained in canu 1.7 assemblies that weren't there in assemblies from previous versions; occasionally they are plasmids, but we are also finding extra repeat copies, subtelomeres etc being retained that weren't present in canu 1.6 assemblies. Is that expected, and is there a way to tweak canu to be less conservative in what it retains? (I don't have a clean example of this right now as we're using read sets that have been re-basecalled with later versions of Albacore, so that could be at fault, but I can prepare a proper example if it would be useful.)

**Comment by peterthorpe5** [comment]
HI John,

I am just a user of the software(nothing to do with development) but I thought I would comment on corOutCoverage. If you dont know your genome size, the default 40X based of the genome size you give it can end up being a lot lower (no kmer genome estimation is not working for my beast - it may be tetra ploid). If you have highly heterozygous data, and/or a polyploid genome (read the FAQ for this bit) adjusting corOutCoverage=200 correctedErrorRate=0.15 can help. 

So basically is 40X enough for what you want?

cheers,

Pete

**Comment by skoren** [comment]
Right @peterthorpe5 is correct. The rescue will catch things poorly represented by the 40x selected by default. Think, if they have significantly different k-mers. This works for things like plasmids because they don't look like the chromosome for most of their bases. 

Typically the heterozygosity is not high enough given the high error rates of the raw sequences that one haplotype still "represents" the other haplotype well. I expect if you have very high heterozygosity (5-10%) it will start rescuing the haplotypes as well.

**Comment by johnomics** [comment]
Ah, OK - I might try adjusting correctedErrorRate to absorb haplotypes but retain plasmids. Thanks both!
